### PR TITLE
[Docs] Improve docs on EH stack trace support

### PR DESCRIPTION
**Pull request by aheejin**
This improves docs on stack trace support for exceptions.
- Mentions you also need `-g`/`-gsource-map`
- Mentions stack trace support is available only for Wasm exceptions
- Illustrates how you can examine `stack` property directly

### CODE REVIEW

File: site/source/docs/porting/exceptions.rst [hunk 1]
```diff
@@ -79,14 +83,41 @@ be supported in all WebAssembly engines yet
...
+------------
+
+For :ref:`native Wasm exceptions
+<webassembly-exception-handling-based-support>`, when :ref:`ASSERTIONS
+<debugging-ASSERTIONS>` is enabled, uncaught exceptions will print stack traces
+for debugging. :ref:`ASSERTIONS <debugging-ASSERTIONS>` is enabled by default
+in :ref:`-O0 <emcc-O0>` and disabled in optimized builds (:ref:`-O1 <emcc-O1>`
+and above). You can enable it by passing ``-sASSERTIONS`` to the ``emcc``
+command line in optimized builds as well. To display Wasm function names in
+stack traces, you also need :ref:`-g <emcc-g>` or :ref:`-gsource-map
```
Comment (sbc100) [inline site/source/docs/porting/exceptions.rst:96]: Or `--profiling-funcs`

File: site/source/docs/porting/exceptions.rst [hunk 2]
```diff
@@ -79,14 +83,41 @@ be supported in all WebAssembly engines yet
...
+.. code-block:: javascript
+
+  try {
+    ... // some code that calls WebAssembly
+  } catch (e) {
+    // Do something with e.stack
+    console.log(e.stack);
+  }
+
+Stack traces within Wasm code are not supported in :ref:`JavaScipt-based
```
Comment (sbc100) [inline site/source/docs/porting/exceptions.rst:113]: I'm not sure what you mean here.  Surly you can catch emscripten EH exceptions and see their stack trace can't you?

**Review by sbc100 — APPROVED** [review]

File: site/source/docs/porting/exceptions.rst (hunk 1 above)
Comment (aheejin) [reply site/source/docs/porting/exceptions.rst]: Done

File: site/source/docs/porting/exceptions.rst (hunk 2 above)
Comment (aheejin) [reply site/source/docs/porting/exceptions.rst:113]: I don't think so?

```console
aheejin@aheejin:~/test/stack$ emcc -g -fexceptions test.cpp -o test.js -sENVIRONMENT=shell,node

aheejin@aheejin:~/test/stack$ node --trace-uncaught test.js

/usr/local/google/home/aheejin/test/stack/test.js:144
      throw ex;
      ^
66800
Thrown at:
    at /usr/local/google/home/aheejin/test/stack/test.js:144:7
    at emit (node:events:526:28)
    at node:internal/process/execution:167:25

Node.js v17.5.0

aheejin@aheejin:~/test/stack$ d8 test.js
exiting due to exception: 66800
```
Node only shows the JS part and d8 doesn't show anything.

JS exception library uses `makeThrow`, which just throws a number: https://github.com/emscripten-core/emscripten/blob/90cccdf2798c22b4fa3f82f586a57147750019de/src/parseTools.js#L614

Unless you create an `Error` object, it doesn't have stack traces. But if we make it an `Error`, all JS exception related library functions, based on that the thrown thing is a number (an address, to be precise), wouldn't work.

Comment (kleisauke) [reply site/source/docs/porting/exceptions.rst:113]: FWIW, draft PR #18535 would fix that.

Comment (sbc100) [reply site/source/docs/porting/exceptions.rst:113]: Oh, I didn't realize that throwing a number doesn't give you a stack trace.

File: site/source/docs/porting/exceptions.rst [hunk 3]
```diff
@@ -79,14 +83,41 @@ be supported in all WebAssembly engines yet
...
+
+For :ref:`native Wasm exceptions
+<webassembly-exception-handling-based-support>`, when :ref:`ASSERTIONS
+<debugging-ASSERTIONS>` is enabled, uncaught exceptions will print stack traces
+for debugging. :ref:`ASSERTIONS <debugging-ASSERTIONS>` is enabled by default
+in :ref:`-O0 <emcc-O0>` and disabled in optimized builds (:ref:`-O1 <emcc-O1>`
+and above). You can enable it by passing ``-sASSERTIONS`` to the ``emcc``
+command line in optimized builds as well. To display Wasm function names in
+stack traces, you also need :ref:`-g <emcc-g>`, :ref:`-gsource-map
+<emcc-gsource-map>`, or :ref:`--profiling-funcs <emcc-profiling-funcs>`.
```
Comment (kripken) [inline site/source/docs/porting/exceptions.rst:97]:
```suggestion
stack traces, you also need :ref:`--profiling-funcs <emcc-profiling-funcs>`
(or :ref:`-g <emcc-g>` or :ref:`-gsource-map<emcc-gsource-map>`).
```
(seems best to suggest the method with the least overhead first)

**Review by kripken — APPROVED** [review]

**Pull request merged**